sdgym/datasets.py

```diff
@@ -45,9 +47,24 @@ def _get_dataset_path(dataset, datasets_path, bucket=None, aws_key=None, aws_sec
     return dataset_path
 
 
-def load_dataset(dataset, datasets_path=None, bucket=None, aws_key=None, aws_secret=None):
+def load_dataset(dataset, datasets_path=None, bucket=None, aws_key=None, aws_secret=None,
+                 max_columns=None):
     dataset_path = _get_dataset_path(dataset, datasets_path, bucket, aws_key, aws_secret)
-    metadata = Metadata(str(dataset_path / 'metadata.json'))
+    metadata_file = open(str(dataset_path / 'metadata.json'))
+    metadata_content = json.load(metadata_file)
+    if max_columns:
+        for table in metadata_content['tables']:
```

Comment:
I would avoid 2 things here:
1. Accessing multiple times the same dict element (in this case `tables`). Instead, it is better to capture it in a variable.
2. Iterating over the dictionary keys to just get the corresponding values. Let's iterate over the `values` instead. (If the table name is needed at any point, then iterate over `items` capturing the name and table in a single shot.

```python3
tables = metadata_content['tables']
for table in tables.values():
    fields = table['fields']
    ...
```